graylog2-server/src/main/java/org/graylog2/filters/ExtractorFilter.java

```diff
@@ -75,17 +76,22 @@ public boolean filter(Message msg) {
                 public List<Extractor> call() throws Exception {
                     LOG.debug("Re-loading extractors for input <{}> into cache.", inputId);
 
-                    Input input = inputService.find(inputId);
+                    try {
+                        Input input = inputService.find(inputId);
 
-                    List<Extractor> sorted = Lists.newArrayList(inputService.getExtractors(input));
+                        List<Extractor> sorted = Lists.newArrayList(inputService.getExtractors(input));
```

Comment:
Not exactly changed in this PR, but why don't we return a sorted list in `InputService#getExtractors(Input)`? Should be easier and would prevent unnecessary copies of the list (and in-place sorting of that list).